src/tools/rbd/action/DiskUsage.cc

```diff
@@ -161,10 +163,36 @@ static int do_disk_usage(librbd::RBD &rbd, librados::IoCtx &io_ctx,
                     snap_list.end());
 
     bool found_from_snap = (from_snapname == nullptr);
+    bool found_snap = (snapname == nullptr);
+    bool found_from = (from_snapname == nullptr);
     std::string last_snap_name;
     std::sort(snap_list.begin(), snap_list.end(),
               boost::bind(&librbd::snap_info_t::id, _1) <
                 boost::bind(&librbd::snap_info_t::id, _2));
+    if (!found_snap || !found_from) {
+      for (auto &snap_info : snap_list) {
+        if (!found_snap && snap_info.name == snapname) {
+          snap_id = snap_info.id;
+          found_snap = true;
+        }
+        if (!found_from && snap_info.name == from_snapname) {
+          from_id = snap_info.id;
+          found_from = true;
+        }
+        if (found_snap && found_from) {
+          break;
+        }
+      }
+    }
+    if ((snapname != nullptr && snap_id == 0) ||
+        (from_snapname != nullptr && from_id == 0)) {
+      std::cerr << "specified snapshot is not found." << std::endl;
+      return -ENOENT;
+    }
+    if (snapname != nullptr && (from_id > snap_id)) {
```

Comment:
Nit: perhaps it should be ```>=``` if both ```from_id``` and ```snap_id``` are not ```CEPH_NOSNAP```

Reply:
If `from_id == snap_id` and both are not `CEPH_NOSNAP`, it will return `-EINVAL` at here (it's not what we need, though).

Reply:
It seems invalid to me to allow ```--from-snap A``` and ```--snap A``` -- that's the purpose of just specifying a snap (it shows you just that single snap). For example, you wouldn't run ```rbd export-diff --from-snap A image@A```.